include/rapidjson/writer.h

```diff
@@ -352,7 +352,7 @@ class Writer {
         char* end = internal::dtoa(d, buffer, maxDecimalPlaces_);
         PutReserve(*os_, static_cast<size_t>(end - buffer));
         for (char* p = buffer; p != end; ++p)
-            PutUnsafe(*os_, static_cast<typename TargetEncoding::Ch>(*p));
+            PutUnsafe(*os_, static_cast<typename OutputStream::Ch>(*p));
```

Comment:
The warning triggers when `Steam::Ch` is `wchar_t` (UTF16 iirc) and `TargetEncoding` is `UTF8<>`.